Reserving new board number for Mountain eagle h743

pegro66 opened the pull request:
Reserve Board-ID for Mountain Eagle H743

board info on:
[www.mountain-eagle-electronics.com](http://www.mountain-eagle-electronics.com/)

peterbarker reviewed (changes requested):
This should be a single commit against `board_types.txt`, just adding the board type.

The commit should be titled `board_types.txt: reservie ID for MountainEagleH743`.

Let me know if you want me to do one of those for you.

pegro66 commented:
... maybe you do one for me please - I'm getting annoyed with GitHub-Desktop doing thins I do not want to include ...

//Peter

peterbarker commented:
> ... maybe you do one for me please - I'm getting annoyed with GitHub-Desktop doing thins I do not want to include ...

See https://github.com/ArduPilot/ardupilot/pull/27046

Also see https://ardupilot.org/dev/docs/where-to-get-the-code.html and the documents linked from it ("rebasing" and "cleaning up the commits".

pegro66 commented:
@peterbarker Thanks - that helps !

Now - how to proceed to get the board into the wiki-page ?

I basically followed the process described here :
https://ardupilot.org/dev/docs/porting.html

//Peter

MATEKSYS commented:
looks like most of definitions are copied from MatekH743, even the APJ_BOARD_ID 1013 is copied. :-)

pegro66 commented:
... the MatekH743 was the basis for the hwdef - as proposed on that web page ...

Meanwhile I did some more changes to fit all the pins and buses plus I have the schematics in KiCAD to add it as an open source board

I also updated the board iD to the assigned "1444"

How can I send that package of files ?

//Peter

pegro66 commented:
@MATEKSYS I have a MatekH743 in my T1-Ranger and it works very well there (and I had some experience with its configuration). So thanks for your development.
For the MTX640 it was too big - so I was looking for something with a smaller footprint - here we go ...
//Peter

pegro66 commented:
@peterbarker  ... the MatekH743 was the basis for the hwdef - as proposed on that web page ...

Meanwhile I did some more changes to fit all the pins and buses plus I have the schematics in KiCAD to add it as an open source board

I also updated the board iD to the assigned "1444"

How can I send that package of files ?

//Peter

peterbarker commented:
> How can I send that package of files ?

@pegro66 not sure what you mean by "package of files".  If you are referring to a KiCAD file and supporting files, a PR against https://github.com/ArduPilot/Schematics would be welcome!

peterbarker commented:
@pegro66 the commit list in here needs a good cleanup.  That's rebasing and squashing commits, and putting them into a one-commit-per-directory format.

 @Hwurzburg put together a nice Wiki page on how to do that here: https://ardupilot.org/dev/docs/git-interactive-rebase.html

... but I can do that for you if you would like.

Hwurzburg commented:
Needs readme, images,etc...see https://github.com/ArduPilot/ardupilot/tree/master/libraries/AP_HAL_ChibiOS/hwdef/JHEMCU-H743HD for an example

Hwurzburg reviewed (changes requested):
Needs readme, images,etc...see https://github.com/ArduPilot/ardupilot/tree/master/libraries/AP_HAL_ChibiOS/hwdef/JHEMCU-H743HD for an example

peterbarker commented:
Ping @pegro66 - are you still interested in getting this work merged?

pegro66 commented:
Hello Peter,

absolutely - what do I have to do ?

//Peter





On 11. Sep 2024, at 09:01, Peter Barker ***@***.***> wrote:


Ping @pegro66 <https://github.com/pegro66> - are you still interested in getting this work merged?

—
Reply to this email directly, view it on GitHub <https://github.com/ArduPilot/ardupilot/pull/27044#issuecomment-2342415129>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AEUYXR2YTGATE3VUQPX2B5LZV6I5TAVCNFSM6AAAAABHSJR7RCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDGNBSGQYTKMJSHE>.
You are receiving this because you were mentioned.

peterbarker commented:
> Hello Peter, absolutely - what do I have to do ? //Peter On 11. Sep 2024, at 09:01, Peter Barker ***@***.***> wrote: Ping 

Henry has asked you to supply a README.md and some images so he can populate the Wiki.

He has pointed you at a board which has the desired information already, so "do what it does".

Also note my request for a clean commit list; we have some notes on using git to do that here: https://ardupilot.org/dev/docs/git-interactive-rebase.html - but if you are going to struggle with that Henry or myself could clean that up for you.

If the board design is done you could also consider creating that PR against the schematics repo.

pegro66 commented:
Hello Peter,
I'm a bit slow and totally unfamiliar with git - I have used ClearCase for some 30 years and I'm struggling with this git ...
Anyway - I have created a repository in my GitHub containing all the files that are necessary:

https://github.com/pegro66/Mountain-Eagle-H743.git

If you can give me a pointer on how to "cleanup" the mess - I'm happy to give it a try (but it will be slow as I only have access to git in the early morning of a WE using a VPN as I'm currently working in China) - otherwise I would be happy if you can lend me a helping hand :)

Best regards
Peter

pegro66 commented:
Hello Peter,
I'm a bit slow and totally unfamiliar with git - I have used ClearCase for some 30 years and I'm struggling with this git ...
Anyway - I have created a repository in my GitHub containing all the files that are necessary:

https://github.com/pegro66/Mountain-Eagle-H743.git

If you can give me a pointer on how to "cleanup" the mess - I'm happy to give it a try (but it will be slow as I only have access to git in the early morning of a WE using a VPN as I'm currently working in China) - otherwise I would be happy if you can lend me a helping hand :)

Best regards
Peter

pegro66 commented:
Hello Peter,
I'm a bit slow and totally unfamiliar with git - I have used ClearCase for some 30 years and I'm struggling with this git ...
Anyway - I have created a repository in my GitHub containing all the files that are necessary:

https://github.com/pegro66/Mountain-Eagle-H743.git

If you can give me a pointer on how to "cleanup" the mess - I'm happy to give it a try (but it will be slow as I only have access to git in the early morning of a WE using a VPN as I'm currently working in China) - otherwise I would be happy if you can lend me a helping hand :)

Best regards
Peter

Hwurzburg commented:
@pegro66 Peter, I had some time, so I transferred your files from your private repo to this PR, fixed an LED error that prevented compile, added the bootloader files, squashed, library split and pushed back to this PR...I haven't reviewed the README vs dat files yet, but I should be able to this week...

has this been tested?
can you provide the English translation for the Chinese notes in the image? I can fix those when this gets merged and I make a wiki page for it..

pegro66 commented:
Hello Henry,

thanks for your support !!!

The configuration compiled on my Linux computer - I’m using the boot loader and SW on my Helicopter … but if it didn’t work with you I guess there must some bug - maybe I oversaw a warning

Picture:
I will update them with English instead of the Chinese - will let you know when I have them uploaded.

Best regards
Peter




On 15. Oct 2024, at 21:40, Henry Wurzburg ***@***.***> wrote:


@pegro66 <https://github.com/pegro66> Peter, I had some time, so I transferred your files from your private repo to this PR, fixed an LED error that prevented compile, added the bootloader files, squashed, library split and pushed back to this PR...I haven't reviewed the README vs dat files yet, but I should be able to this week...

has this been tested?
can you provide the English translation for the Chinese notes in the image? I can fix those when this gets merged and I make a wiki page for it..

—
Reply to this email directly, view it on GitHub <https://github.com/ArduPilot/ardupilot/pull/27044#issuecomment-2413950025>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AEUYXR27MWI27IE23FFIKN3Z3ULLLAVCNFSM6AAAAABHSJR7RCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMJTHE2TAMBSGU>.
You are receiving this because you were mentioned.

Hwurzburg commented on libraries/AP_HAL_ChibiOS/hwdef/MountainEagleH743/hwdef.dat:
```diff
@@ -0,0 +1,195 @@
+# LED
+# LED0-orange
+# LED1-green
+PE9 LED0 OUTPUT LOW GPIO(90) # orange
```
in addition to get it to compile, you will need to add this new define to get the bd leds to work```suggestion
define AP_NOTIFY_GPIO_LED_2_ENABLED 1
PE9 LED0 OUTPUT LOW GPIO(90) # orange
```

pegro66 commented:
@Hwurzburg
Hello Henry,
finally found some time to do the proposed changes
uploaded everything to

https://github.com/pegro66/Mountain-Eagle-H743 <https://github.com/pegro66/Mountain-Eagle-H743>

Thanks for your support !
Peter



On 15. Oct 2024, at 21:40, Henry Wurzburg ***@***.***> wrote:


@pegro66 <https://github.com/pegro66> Peter, I had some time, so I transferred your files from your private repo to this PR, fixed an LED error that prevented compile, added the bootloader files, squashed, library split and pushed back to this PR...I haven't reviewed the README vs dat files yet, but I should be able to this week...

has this been tested?
can you provide the English translation for the Chinese notes in the image? I can fix those when this gets merged and I make a wiki page for it..

—
Reply to this email directly, view it on GitHub <https://github.com/ArduPilot/ardupilot/pull/27044#issuecomment-2413950025>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AEUYXR27MWI27IE23FFIKN3Z3ULLLAVCNFSM6AAAAABHSJR7RCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIMJTHE2TAMBSGU>.
You are receiving this because you were mentioned.

Hwurzburg commented:
@pegro66 are you going to push those changes to this PR?

pegro66 commented:
… absolutely … the question is how ?




On 3. Dec 2024, at 00:06, Henry Wurzburg ***@***.***> wrote:


@pegro66 <https://github.com/pegro66> are you going to push those changes to this PR?

—
Reply to this email directly, view it on GitHub <https://github.com/ArduPilot/ardupilot/pull/27044#issuecomment-2511947458>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AEUYXR4YEPR6BU5XK4PFTMT2DSASHAVCNFSM6AAAAABHSJR7RCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKMJRHE2DONBVHA>.
You are receiving this because you were mentioned.

peterbarker commented:
> … absolutely … the question is how ? On 3. Dec 2024, at 00:06, Henry Wurzburg ***@***.***> wrote: @pegro66 <https://

You need to force-push your PR branch.  See https://ardupilot.org/dev/docs/git-rebase.html

pegro66 commented:
Hello Henry,we close it - my guy in China has gone silent.Best regards PeterOn 14. Aug 2025, at 15:10, Henry Wurzburg ***@***.***> wrote:﻿Hwurzburg left a comment (ArduPilot/ardupilot#27044)
this has been active for a long time...shall we close? @peterbarker

—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>
